### PR TITLE
[DVCSMP-3656] On-boarding Fibaro Flood Sensor ZW5

### PR DESCRIPTION
**Pull request by sagar-trility**

### CODE REVIEW

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 1]
```diff
@@ -1,300 +1,531 @@
...
+    log.debug "zWaveProtocolSubVersion: ${cmd.zWaveProtocolSubVersion}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+    log.debug "BatteryReport"
+    log.debug "cmd: "+cmd
+    log.debug "location: "+location
+    def timeDate = location.timeZone ? new Date().format("yyyy MMM dd EEE h:mm:ss a", location.timeZone) : new Date().format("yyyy MMM dd EEE h:mm:ss")
+    //def timeDate = new Date().format("yyyy MMM dd EEE HH:mm:ss", location.timeZone)
+    sendEvent(name: "battery", value: cmd.batteryLevel)
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: there's old logic in the handler that has been removed that needs to treat the special z-wave value of 255 as "low battery"

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 2]
```diff
@@ -1,300 +1,531 @@
...
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+    log.debug "SensorMultilevelReport"
+    def tempValue
+    def map = [:]
+    if (cmd.sensorType == 1) {
+        // temperature
+        def cmdScale = cmd.scale == 1 ? "F" : "C"
+        tempValue = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, 1)
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: confused about why two identical variables are needed here

Comment (sagar-trility) [reply devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: Change has been made. Before I update the PR I will have to run a test to make sure.

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 3]
```diff
@@ -1,301 +1,531 @@
...
+
+private Map cmdVersions() {
+    [0x31: 5, 0x56: 1, 0x71: 3, 0x72: 2, 0x80: 1, 0x84: 2, 0x85: 2, 0x86: 1, 0x98: 1]
+}
+
+def configure() {
+    state.lastAlarmDate = "-"
+    def cmds = []
+    sendEvent(name: "water", value: "dry", displayed: "true")
+    cmds += zwave.wakeUpV2.wakeUpIntervalSet(seconds: 21600, nodeid: zwaveHubNodeId)//FGFS' default wake up interval
```
Comment (ALamchaS) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: For sleepy devices we usually set checkInterval to value higher than 2 times wakeup interval, so checkInterval could be set to something like 2 * 21600 +  6 * 60 (2 * wakeup interval + 6 minutes)

Currently you set it in line 107:
sendEvent(name: "checkInterval", value: 28800, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 4]
```diff
@@ -1,301 +1,531 @@
...
-		crc16(cmd)
-	} else {
-		secure(cmd)
-	}
-}
+/**
+ *  Fibaro Flood Sensor ZW5
+ */
+metadata {
+    definition(name: "Fibaro Flood Sensor ZW5", namespace: "FibarGroup", author: "Fibar Group") {
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: keep ocf device type

**Comment by sagar-trility** [comment]
@greens PR has been updated, please review. Thank You.

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 5]
```diff
@@ -1,338 +1,346 @@
...
-    //def timeDate = new Date().format("yyyy MMM dd EEE HH:mm:ss", location.timeZone)
-    sendEvent(name: "battery", value: cmd.batteryLevel)
-    sendEvent(name: "batteryStatus", value: "Battery: $cmd.batteryLevel%\n($timeDate)")
+	log.debug "BatteryReport"
+	log.debug "cmd: "+cmd
+	log.debug "location: "+location
+	def timeDate = location.timeZone ? new Date().format("yyyy MMM dd EEE h:mm:ss a", location.timeZone) : new Date().format("yyyy MMM dd EEE h:mm:ss")
+	def value = cmd.batteryLevel
+	def desc = "Curent battery level"
+	def isStateChange = false
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: This variable should be left to the internal logic of `sendEvent` and `createEvent` unless it's the special case of low battery.

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 6]
```diff
@@ -1,301 +1,539 @@
...
+private multiStatusEvent(String statusValue, boolean force = false, boolean display = false) {
+	if (!device.currentValue("multiStatus")?.contains("Sync") || device.currentValue("multiStatus") == "Sync OK." || force) {
+		sendEvent(name: "multiStatus", value: statusValue, descriptionText: statusValue, displayed: display)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	log.debug "SensorMultilevelReport"
+	def tempValue
+	def map = [:]
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: I think this map was intended to be used, but it isn't.

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 7]
```diff
@@ -1,301 +1,539 @@
...
+				map.name = "water"
+				map.value = "dry"
+				map.descriptionText = "${device.displayName} is ${map.value}"
+				multiStatusEvent(alarmInfo + state.lastAlarmDate)
+				break
+		}
+	} else if (cmd.notificationType == 7) {
+		switch (cmd.event) {
+			case 0:
+				map.descriptionText = "${device.displayName}: tamper alarm has been deactivated"
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: Just setting the description text is probably not what was intended here. I think the tamper events should be put in the map, while the batteryStatus event should be sent out of band.

**Comment by sagar-trility** [comment]
@greens This is ready for review.

File: devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy [hunk 8]
```diff
@@ -1,346 +1,342 @@
...
-	sendEvent(name: "batteryStatus", value: "Battery: $cmd.batteryLevel%\n($timeDate)")
+    log.debug "BatteryReport"
+    log.debug "cmd: "+cmd
+    log.debug "location: "+location
+    def timeDate = location.timeZone ? new Date().format("yyyy MMM dd EEE h:mm:ss a", location.timeZone) : new Date().format("yyyy MMM dd EEE h:mm:ss")
+
+    if (value == 0xFF) {  // Special value for low battery alert
+        sendEvent(name: "battery", value: 1, descriptionText: "${device.displayName} has a low battery", isStateChange: true)
+    } else {
+        sendEvent(name: "battery", value: cmd.batterLevel, descriptionText: "Curent battery level")
```
Comment (greens) [inline devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy]: typo in `cmd.batteryLevel` and "Current"

**Comment by sagar-trility** [comment]
@greens PR has been updated and ready for review.

**Review by greens — APPROVED** [review]

**Pull request merged**